spec/formatting.md

```diff
@@ -834,6 +834,21 @@ These are divided into the following categories:
     > {{{$var} cannot be redefined. {$var2} cannot refer to itself}}
     > ```
 
+  - A **Forward Reference error** occurs when a _declaration_ _expression_ includes a _variable_
+    that is declared after it in the _message_.
```

Comment:
Is it truly a "forward reference error" or it is actually a species of "duplicate name"?

That is, with `.local $foo = {$bar} .local $bar = {something}`, the first `$bar` might be resolved from the inputs (and that could fail), making the second `local` an attempt at reassignment.

Reply:
Which confusion is why we want this to be an error. I'd be fine describing this either way, as long as we do include the error.